Reset password

Ammar-IT99 opened the pull request:
feat(auth): Implement forget password and reset password functionality.

This commit introduces the ability for users to reset their password if they forget it. 
- Added `ForgetPasswordUseCase` and `ResetPasswordUseCase` to handle the business logic.
- Implemented `AuthViewModel` to manage the UI state and interact with the use cases.
- Created `ForgetPasswordScreen`, `OtpScreen`, and `ResetPasswordScreen` for the user interface flow.
- Added necessary network requests to the `WebServices` class.
- Updated dependency injection configuration.

alkharashi20 committed:
feat(SignUpScreen): Add Logic For SignUP Screen.

Ammar-IT99 committed:
feat(SignUpScreen,LoginScreen,ForgetPasswordScreen): update UI and add validation for text fields.

Ammar-IT99 committed:
feat(SignUpScreen,LoginScreen,ForgetPasswordScreen): update UI and add validation for text fields.

Ammar-IT99 committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/logic_signup' into reset_password

# Conflicts:
#	lib/core/utils/resources/main_text_field.dart
#	lib/core/utils/resources/validator_manager.dart
#	lib/data/web_services/WebServices.dart
#	lib/data/web_services/WebServices.g.dart
#	lib/di/injetible_intinalizer.config.dart
#	lib/presentation/auth/cubit/auth_states.dart
#	lib/presentation/auth/cubit/auth_view_model.dart
#	lib/presentation/auth/screens/login_screen.dart
#	lib/presentation/auth/screens/signup_screen.dart
#	pubspec.lock
#	pubspec.yaml

Ammar-IT99 committed:
feat(development): Refactor AuthViewModel and remove signup-related files.

Ammar-IT99 committed:
feat(auth): Implement forget password and reset password functionality.

This commit introduces the ability for users to reset their password if they forget it.
- Added `ForgetPasswordUseCase` and `ResetPasswordUseCase` to handle the business logic.
- Implemented `AuthViewModel` to manage the UI state and interact with the use cases.
- Created `ForgetPasswordScreen`, `OtpScreen`, and `ResetPasswordScreen` for the user interface flow.
- Added necessary network requests to the `WebServices` class.
- Updated dependency injection configuration.
This closes #1.

Ammar-IT99 committed:
Merge branch 'development' into reset_password

john-safwat commented on lib/core/utils/resources/custom_elevated_button.dart:
```diff
@@ -35,7 +35,7 @@ class CustomElevatedButton extends StatelessWidget {
               ? const StadiumBorder()
               : RoundedRectangleBorder(
-                  borderRadius: BorderRadius.circular(17.r)),
+              borderRadius: BorderRadius.circular(17.r)),
```
spaces and borders should be Multiples of 8 for design consistency

john-safwat commented on lib/data/data_source/remote_data_source/auth_remote_data_source/auth_remote_data_source_implementation.dart:
```diff
@@ -35,4 +37,21 @@ class AuthRemoteDataSourceImplementation
+          email: request.email,
+          newPassword: request.newPassword,)
+      );
+      print("🔵 API Response: ${response.toJson()}"); // طباعة الداتا المسترجعة
```
avoid using print in production code instead use debug print

john-safwat commented on lib/data/data_source/remote_data_source/auth_remote_data_source/signup/signup_remote_data_source_contract.dart:
```diff
@@ -0,0 +1,8 @@
+import 'package:flowery/domain/entity/auth_entity/signup/signup_entity.dart';
+
+abstract class SignupRemoteDataSourceContract{
+Future<ApiResult<SignupEntity>>signup(String firstName , String lastName , String email,String password,String rePassword ,String phone,String gender);
```
create entity or model for encapsulating the data don't pass it like this for future updates

john-safwat commented on lib/data/data_source/remote_data_source/auth_remote_data_source/signup/signup_remote_data_source_impl.dart:
```diff
@@ -0,0 +1,38 @@
+  SignUpRemoteDataSourceImpl({required this.webServices});
+
+  @override
+  Future<ApiResult<SignupResponseDto>> signup(String firstName, String lastName, String email, String password, String rePassword, String phone, String gender) async {
+    return executeApi<SignupResponseDto>(()async {
+return    await webServices.getpost({
+    "firstName": firstName,
+    "lastName": lastName,
+    "email": email,
+    "password": password,
+    "rePassword": rePassword,
+    "phone": phone,
+    "gender": gender
+  });
```
replace this parameters with model and function to json to create the request body

john-safwat commented on lib/data/repository_implementation/auth_repository_implementation/signup_repository_impl.dart:
```diff
@@ -0,0 +1,22 @@
+import 'package:flowery/domain/repository_contract/auth_repository_contract/signup/signup_repository.dart';
+import 'package:injectable/injectable.dart';
+@Injectable(as:SignupRepository )
+class SignUpRepositoryImpl implements SignupRepository{
+  SignupRemoteDataSourceContract signupRemoteDataSourceContract;
+  SignUpRepositoryImpl({required this.signupRemoteDataSourceContract});
+  @override
+  Future<ApiResult<SignupEntity>> SignUp(String firstName , String lastName , String email,String password,String rePassword ,String phone,String gender) {
+
+
+  return signupRemoteDataSourceContract.signup(firstName, lastName, email, password, rePassword, phone, gender);
+
```
replace with object of model

john-safwat commented on lib/domain/entity/auth_entity/signup/signup_entity.dart:
```diff
@@ -0,0 +1,48 @@
+import 'package:json_annotation/json_annotation.dart';
+@JsonSerializable()
```
no need for serialization in entity

john-safwat commented on lib/domain/repository_contract/auth_repository_contract/signup/signup_repository.dart:
```diff
@@ -0,0 +1,8 @@
+import 'package:flowery/domain/entity/auth_entity/signup/signup_entity.dart';
+
+abstract class SignupRepository{
+  Future<ApiResult<SignupEntity>>SignUp(String firstName , String lastName , String email,String password,String rePassword ,String phone,String gender);
+
```
replace with object

john-safwat commented on lib/domain/use_case/auth_use_case/signup_use_case.dart:
```diff
@@ -0,0 +1,16 @@
+class SignUpUseCase{
+  final SignupRepository signupRepository;
+  SignUpUseCase({required this.signupRepository});
+  Future<ApiResult<SignupEntity>>SignUp(String firstName , String lastName , String email,String password,String rePassword ,String phone,String gender){
+
+return signupRepository.SignUp(firstName, lastName, email, password, rePassword, phone, gender);
```
replace with object

john-safwat commented on lib/presentation/auth/cubit/auth_state.dart:
state class should hold cubit states no the data here you are using it to hold the data

john-safwat commented on lib/presentation/auth/screens/forget_password_screen.dart:
```diff
@@ -74,7 +77,7 @@ class ForgetPassword extends StatelessWidget {
                   CustomElevatedButton(
                     label: "Confirm",
```
don use hard coded string in your code

john-safwat commented on lib/presentation/auth/screens/reset_password_screen.dart:
```diff
@@ -0,0 +1,153 @@
+      BlocConsumer<AuthViewModel, AuthState>(
+        listener: (context, state) {
+          if (state.status == Status.loading) {
+            DialogUtils.showLoadingDialog(context, message: 'Loading...');
+          } else {
+            DialogUtils.hideLoadingDialog(context);
+          }
+
+          if (state.status == Status.success&&state.resetPasswordResponse!=null) {
+            DialogUtils.showMessageDialog(
+              context,
+              message: state.resetPasswordResponse?.message?? 'Password reset successfully',
+              posTitle: 'OK',
+              posAction: () {
+                Navigator.pushNamedAndRemoveUntil(
+                  context,
+                  RoutesNames.loginScreen,
+                      (route) => false,
+                );
+              },
+            );
+          } else if (state.status == Status.error&&state.exception  != null) {
+            DialogUtils.showMessageDialog(
+              context,
+              message: state.resetPasswordResponse?.error??"",
+              posTitle: 'Try Again',
+            );
+          }
```
don't use hard coded strings in your code

john-safwat commented on lib/presentation/auth/screens/reset_password_screen.dart:
```diff
@@ -0,0 +1,153 @@
+                      height: 50,
+                      child:
+                      CustomElevatedButton(
+                        label: "Continue",
```
don't use hard coded string in your code

john-safwat commented on lib/domain/entity/auth_entity/signup/error.dart:
don't add empty files to your project add files only when needed

Ammar-IT99 committed:
feat(auth): Implement forget password and reset password functionality.

This commit introduces the ability for users to reset their password if they forget it.
- Added `ForgetPasswordUseCase` and `ResetPasswordUseCase` to handle the business logic.
- Implemented `AuthViewModel` to manage the UI state and interact with the use cases.
- Created `ForgetPasswordScreen`, `OtpScreen`, and `ResetPasswordScreen` for the user interface flow.
- Added necessary network requests to the `WebServices` class.
- Updated dependency injection configuration.
This closes #1.

Ammar-IT99 committed:
Delete auth related files under signup directory.

Ammar-IT99 reviewed (commented):
feat / delete empty file

Ammar-IT99 committed:
feat(auth): Add string manager in reset and forget password screen.
